File: server/src/main/java/com/yuy/eduflow/teacher/Teacher.java
package com.yuy.eduflow.teacher;

import com.yuy.eduflow.enums.ActiveStatus;
import java.time.LocalDateTime;
import lombok.Data;

@Data
public class Teacher {
	private Long id;
	private String employeeNo;
	private String password;
	private String role;
	private String name;
	private String department;
	private String title;
	private Integer maxWeeklyHours;
    private ActiveStatus status;
	private LocalDateTime createdAt;
	private LocalDateTime updatedAt;
}

File: web/src/types/teacher.ts
import type { ActiveStatus } from './status'

export type TeacherRole = 'TEACHER' | 'ADMIN'

export interface Teacher {
  id: number
  employeeNo: string
  password?: string
  role: TeacherRole
  name: string
  department?: string
  title?: string
  maxWeeklyHours: number
  status: ActiveStatus
  createdAt?: string
  updatedAt?: string
  displayName?: string
}
